examples/lock-app/cc13x2_26x2/BUILD.gn

```diff
@@ -0,0 +1,161 @@
+    defines += [ "BUILD_RELEASE=0" ]
+  } else {
+    defines += [ "BUILD_RELEASE=1" ]
+  }
+}
+
+source_set("openthread_platform") {
+  configs -= [ "//build/config/compiler:std_default" ]
+  configs += [ ":sdk_posix_config" ]
+  defines = [ "NVOCMP_POSIX_MUTEX" ]
+  cflags = [
+    "-Wno-int-conversion",
+    "-Wno-address-of-packed-member",
+  ]
+  include_dirs = [
+    "${chip_root}/third_party/openthread/repo/examples/platforms",
+    "${chip_root}/third_party/openthread/repo/src/core",
+    "${ti_simplelink_sdk_root}/source/ti/devices/cc13x2_cc26x2",
+  ]
+  public_deps = [
+    ":sdk",
+    ":sysconfig",
+    "${chip_root}/third_party/openthread/repo:libopenthread-cli-mtd",
+    "${chip_root}/third_party/openthread/repo:libopenthread-mtd",
+  ]
+  sources = [
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/alarm.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/alarm_micro.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/crypto/aes_alt.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/crypto/ecjpake_alt.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/crypto/sha256_alt.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/diag.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/entropy.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/misc.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/nv/crc.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/nv/nvocmp.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/radio.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/settings.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/system.c",
+    "${ti_simplelink_sdk_root}/examples/rtos/${ti_simplelink_board}/thread/cli_mtd/platform/uart.c",
+  ]
+}
+
+ti_sysconfig("sysconfig") {
+  sources = [ "${project_dir}/chip.syscfg" ]
+  outputs = [
+    "ti_devices_config.c",
+    "ti_radio_config.c",
+    "ti_radio_config.h",
+    "ti_drivers_config.c",
+    "ti_drivers_config.h",
+
+    # disabled until upstream generation is aligned
+    #"tiop_config.h",
+    #"tiop_config.c",
+
+    # not traditional source files
+    #"ti_utils_build_linker.cmd.genlibs",
+    #"syscfg_c.rov.xs",
+    #"ti_utils_runtime_model.gv",
+    #"ti_utils_runtime_Makefile",
+  ]
+}
+
+ti_simplelink_executable("lock_app") {
+  include_dirs = [
+    "${project_dir}",
+    "${project_dir}/main",
+    "${project_dir}/main/include",
+    "${chip_root}/src/include",
+    "${chip_root}/src/lib",
```

Comment:
Something wrong here - why do we need to duplicate the CHIP include_dirs? They should be added automatically by depending on libCHIP.

Reply:
This spurred me to go look into the actual build dependencies. These includes were from when I was trying to figure out how everything built and just wanted to include everything. I should have done a better second round review.

The dependency tree should be pretty clean now.